### PR TITLE
Use official `llvm-project` monorepo, expand LLDB versions check

### LINKED ISSUE #350: Install Fails for LLDB-10
FYI - It looks like https://github.com/llvm-mirror/lldb.git doesn't have a release_100 branch that maps to lldb-10.
So the error below gets thrown.

Does it need a patch to warn the user in a friendlier way?
Not 100% about the release cadence on the llvm mirror

```
Cloning lldb release_100 into /usr/local/lib/node_modules/llnode/lldb-10.0
Cloning into '/usr/local/lib/node_modules/llnode/lldb-10.0'...
warning: Could not find remote branch release_100 to clone.
fatal: Remote branch release_100 not found in upstream origin
child_process.js:651
    throw err;
    ^

Error: Command failed: git clone --depth 1 --branch release_100 https://github.com/llvm-mirror/lldb.git /usr/local/lib/node_modules/llnode/lldb-10.0
    at checkExecSyncError (child_process.js:630:11)
    at Object.execFileSync (child_process.js:648:15)
    at Object.cloneHeaders (/usr/local/lib/node_modules/llnode/scripts/lldb.js:47:19)
    at configureInstallation (/usr/local/lib/node_modules/llnode/scripts/configure.js:89:25)
    at main (/usr/local/lib/node_modules/llnode/scripts/configure.js:12:34)
    at Object.<anonymous> (/usr/local/lib/node_modules/llnode/scripts/configure.js:20:1)
    at Module._compile (internal/modules/cjs/loader.js:1158:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
    at Module.load (internal/modules/cjs/loader.js:1002:32)
    at Function.Module._load (internal/modules/cjs/loader.js:901:14) {
  status: 128,
  signal: null,
  output: [ null, null, null ],
  pid: 3484,
  stdout: null,
  stderr: null
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! llnode@3.0.0 preinstall: `node scripts/configure.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the llnode@3.0.0 preinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

### PR DESCRIPTION
**Pull request by trxcllnt**
* Fetch headers from the official [llvm-project](https://github.com/llvm/llvm-project) instead of the deprecated [llvm-mirror/lldb](https://github.com/llvm-mirror/lldb.git) repo.
* Check for LLDB versions `20.0 - 1.0.0` on Linux so we don't have to update LLDB versions for a few years.

Fixes https://github.com/nodejs/llnode/issues/350

### CODE REVIEW

**Comment by simllll** [comment]
Looking forward to this.. can we use this fork in the meantime somehow? is it published somewhere?

**Comment by trxcllnt** [comment]
@simllll it's not published anywhere, but you can see how we're installing it in our development container here:
* [Install the LLDB version you'd like to use](https://github.com/rapidsai/node-rapids/blob/main/dockerfiles/devel.Dockerfile#L43-L44)
* [Clone my fork and manually install `llnode`](https://github.com/rapidsai/node-rapids/blob/main/dockerfiles/devel.Dockerfile#L160-L163)

**Comment by No9** [comment]
Tested this today and it worked as expected 
@nodejs/diagnostics what can I do to help land this?

**Review by mhdawson — APPROVED** [review]
LGTM.

**Comment by mhdawson** [comment]
@mmarchini any chance you can help with the CI issues?

**Comment by No9** [comment]
Hey @mmarchini 
Did you start working on the PR to [actions/setup-node](https://github.com/actions/setup-node)? 
If you haven't I can look at picking it up.

From what I can see it's these two commits that need reintegrating but let me know if there is anything else or hidden gotchas.
https://github.com/mmarchini/setup-node/commit/6a399da24b33138ea0de9981a95b7e7123f7b1dd 
https://github.com/mmarchini/setup-node/commit/772e9574c06048dd733148cd1cb8052c1395726b 

If they don't accept it we can kick of this process
https://github.com/nodejs/admin/blob/main/transfer-repo-into-the-org.md

**Comment by No9** [comment]
Update:  I've merged the changes from https://github.com/mmarchini/setup-node/commits/mirror 
into https://github.com/No9/setup-node/tree/mirror 
I've also applied it to my llnode fork and reapplied the v15 settings.
https://github.com/No9/llnode/runs/4206002518?check_suite_focus=true
The change seems to be working as it's picking up the mirror code and runs the tests.
However the change is failing due to `verifyObject` no longer working in 14 and I think this is reported here 
https://github.com/nodejs/llnode/issues/375 
I'll continue to work on the failing test.

**Comment by No9** [comment]
Update: I've raised this PR to land a green build https://github.com/nodejs/llnode/pull/398 that we can use as a base to land this.
This PR should probably also add llvm 10 and 11 to the test definitions 
https://github.com/nodejs/llnode/blob/main/.github/workflows/push.yml#L32

**Comment by trxcllnt** [comment]
> @No9 This PR should probably also add llvm 10 and 11 to the test definitions

Will do!

**Comment by No9** [comment]
Hey @trxcllnt 
I've picked up merge rights this week and pulled in the CI fix that this was dependent on.
I know it's been a while but do you want to rebase and then we can get this merged?
I understand we've lost some momentum but it would be great to get it in as it's an important update.

**Comment by trxcllnt** [comment]
@No9 Sure, I'd be happy to!

**Comment by trxcllnt** [comment]
Seems like some LLVM versions aren't packaged by Ubuntu. Should I switch the CI runner to use the LLVM apt repositories listed at https://apt.llvm.org/?

**Comment by trxcllnt** [comment]
@No9 it looks like there's a few specific combos of `matrix.os` + `matrix.llvm` that don't have packages in `universe`, so I'll update the job to only use the LLVM repos for those.

**Comment by No9** [comment]
Hey @trxcllnt 
Took a look at the failing tests tonight.
Looks like later versions of lldb are now returning more lines than expected and these are being truncated.
Also the content of the response has changed.

So lldb 10 to 14 is returning
```
error: expression failed to parse:
error: <user expression 16>:1:1: use of undeclared identifier 'invalid_expr'
invalid_expr
```
Instead of 
```
error: error: use of undeclared identifier 'invalid_expr'
```
This is compounded by the logic in flush truncating lines on carriage returns 
https://github.com/nodejs/llnode/blob/main/test/common.js#L60 
So for 10-14 the only line that is being returned to the test is the ambiguous
```
error: expression failed to parse:
```

I've updated my local version of flush to make an exception for this case 
```
      let line = '';
      let index = 0;

      let inv = buf.indexOf('invalid_expr');
      if (inv !== -1) { 
          line = buf;
          index = buf.length;
        } else {
      
        index = buf.indexOf('\n');

        if (index === -1)
          break;
        
        line = buf.slice(0, index);
      }

      buf = buf.slice(index + 1);
```
And also updated the test to just match on `use of undeclared identifier 'invalid_expr'`
https://github.com/nodejs/llnode/blob/main/test/plugin/inspect-test.js#L661
https://github.com/nodejs/llnode/blob/main/test/plugin/scan-test.js#L30 
I don't think this impacts the integrity of the tests and seems to work.

Very conscious that you've put a lot of work in to this so wanted to give you the option of applying something similar to this PR.
If you don't have the bandwidth and want me to do it I'm also happy to pick it up.

**Comment by trxcllnt** [comment]
@No9 feel free to push any fixes directly to this branch! Lemme know if you need access. Whatever we need to do to get this passing.

**Comment by No9** [comment]
Given this PR has been open for 16 months and the last major change was 6 weeks ago and we have an approval I am going to merge this.
@trxcllnt Thank you so much for all your work if you are ever in Dublin Ireland hit me up for a [insert favourite drink]

**Pull request merged**
